Tests/Client/RequestRecorderURLProtocol.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  MockURLProtocol.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import XCTest
+
+/// This URLProtocol subclass allows to intercept the network communication
+/// and provides the latest network request made.
+class RequestRecorderURLProtocol: URLProtocol {
```

Comment:
At first I thought the `Protocol` in name refers to Swift protocols, then I started reading up and realized it refers to URL/Networking protocols 🤦 So no renaming suggestion from me (which was my initial reaction)